src/App.tsx

```diff
@@ -1,15 +1,36 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { ApplicationState, ConnectedReduxProps } from './configureStore';
 import logo from './logo.svg';
 import styles from './App.module.scss';
+import { ExampleState, toggleOn, toggleOff } from './reducers/example';
+
+interface Props {
```

Comment:
There's a pattern presented in https://medium.com/@resir014/redux-4-typescript-2-9-a-type-safe-approach-7f073917b803 which uses a `PropsFromState` interface to define any props coming from `mapStateToProps`, which can then be intersectioned with any props specific to the component. In this example it would be overkill, but I thought it to be a pretty nice pattern, making it very obvious exactly from whence props were originating. When I first saw it I thought it seemed nicer than our `InternalProps` that we use in addons-frontend, which could include props from multiple sources.

What do you think about adopting a convention like that?

Reply:
Yeah, I think we should adopt the `PropsFromState` idea. I switched to that. It's really just for code readability, not for type hinting (which is what we needed it for in Flow). TypeScript can already figure out mapped props automatically.

Reply:
I thought adopting TS was also about typing less useless code? I don't really want to write code that "TypeScript can already figure out".

Reply:
Well, I think it's nice as documentation to glance at `PropsFromState` and know *immediately* that those are props I should not set. TypeScript would give you an error if you try to set it, anyway, but I think it's still nice for readability.

Reply:
Ok, I don't necessarily agree because, if the language is smart and actually better than we expected, I don't think we should be too verbose, mostly because we were doing that to solve a Flow limitation before (in this case, because I believe we would not have introduced `InternalProps` if Flow was behaving like TS here).

Anyway, let's see how it goes first.